chapters/classes.tex

```diff
@@ -217,17 +217,31 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
 
-Variables declared with the \lstinline!flow! or the \lstinline!stream! type prefix shall be a subtype of \lstinline!Real!.
-
-Type prefixes (that is, \lstinline!flow!, \lstinline!stream!, \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant!, \lstinline!input!, \lstinline!output!) shall only be applied for type,
-record and connector components -- see also record specialized class, \cref{specialized-classes}.
+Type prefixes (that is, \lstinline!flow!, \lstinline!stream!, \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant!, \lstinline!input!, \lstinline!output!) shall only be applied for type, record, operator record, and connector components -- see also record specialized class, \cref{specialized-classes}.
+This is further restricted below; some of these combinations of type prefixes and specialized classes are not legal.
 
 An exception is \lstinline!input! for components whose type is of the special class function type (these can only be used for function formal parameters and has special semantics, see \cref{functional-input-arguments-to-functions}), and the \lstinline!input! prefix is not applied to the elements of the component and is allowed even if the elements have \lstinline!input! or \lstinline!output! prefix.
 
 In addition, instances of classes extending from \lstinline!ExternalObject! may have type prefixes \lstinline!parameter! and \lstinline!constant!, and in functions also type prefixes
 \lstinline!input! and \lstinline!output!, see \cref{external-objects}.
 
-The type prefixes \lstinline!flow!, \lstinline!stream!, \lstinline!input! and \lstinline!output! of a structured component (except as described above) are also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct; e.g.\ the \lstinline!flow! prefix can be used on a record component and all the record elements will generate zero-sum equations, even if elements of a record shall not be declared with the \lstinline!flow! prefix).  When any of the type prefixes \lstinline!flow!, \lstinline!stream!, \lstinline!input! and \lstinline!output! are applied for a structured component, no element of the component may have any of these type prefixes.  The corresponding rules for the type prefixes \lstinline!discrete!, \lstinline!parameter! and \lstinline!constant! are described in \cref{variability-of-structured-entities} for structured components.
+Variables declared with the \lstinline!stream! type prefix shall be a subtype of \lstinline!Real!.
+This is further restricted in \cref{definition-of-stream-connectors}.
+
+The type prefix \lstinline!flow! of a component that is not a primitive element (see \cref{primitive-elements}), is also applied to the elements of the component (this is done after verifying that the type prefixes occurring on elements of the component are correct).
```

Comment:
What is meant by "the type prefixes occurring on elements of the components are correct"?

Reply:
It was present before this PR, and to me it indicates that we first check that you don't have an invalid type prefix on the component - before "overriding" it.

Reply:
It would make sense to me that the "correct" would mean satisfying the restriction that is now on line 237:
>When any of the type prefixes \lstinline!flow!, \lstinline!input! and \lstinline!output! are applied for a structured component, no element of the component may have any of these type prefixes, nor can they have \lstinline!stream! prefix.

Are there any other restrictions? I find it confusing that we refer to the restrictions as if they are already known. But if we are talking about restrictions on line 237 we haven't introduced them yet.

But, you are right, this has existed before this PR. So maybe we need to open a separate issue or PR for it?

Reply:
I tried to create a ticket for the issue but the text is changed so much by this PR, it is hard to base it off the master the way it is now. So I'm approving this PR now and will create a ticket once it is merged.